src/components/AddUsersModal/index.jsx

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm, SubmissionError } from 'redux-form';
+import { Button, Icon, Modal } from '@edx/paragon';
+
+import H3 from '../H3';
+import TextAreaAutoSize from '../TextAreaAutoSize';
+import StatusAlert from '../StatusAlert';
+import FileInput from '../FileInput';
+import {
+  validateEmailTemplateFields,
+  validateEmailAddresses,
+  validateEmailAddressesFields,
+  mergeErrors,
+} from '../../utils';
+
+class AddUsersModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.errorMessageRef = React.createRef();
+    this.modalRef = React.createRef();
+
+    this.validateFormData = this.validateFormData.bind(this);
+    this.handleModalSubmit = this.handleModalSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      enterpriseId,
+      fetchSubscriptionDetails,
+    } = this.props;
+
+    if (enterpriseId) {
+      fetchSubscriptionDetails(enterpriseId);
```

Comment:
Curious why we need to re-fetch the subscription details when the "Add Users" modal mounts?

Reply:
@adamstankiewicz The reason I put the API call here is that subscription count will change every time we assign new licenses, and currently, this information is only displayed inside the modal. 
If you think it would be better to move the API call inside the subscription details page, let me know, I will move it there and receive the subscription count as a prop.

Reply:
Done, Using the subscription details from the context